src/cascadia/TerminalApp/HighlightedTextControl.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "winrt/Windows.UI.Xaml.Interop.h"
+#include "HighlightedTextControl.h"
+
+#include "HighlightedTextControl.g.cpp"
+#include "HighlightedTextSegment.g.cpp"
+#include "HighlightedText.g.cpp"
+
+using namespace winrt;
+using namespace winrt::TerminalApp;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::UI::Text;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Microsoft::Terminal::Settings::Model;
+
+namespace winrt::TerminalApp::implementation
+{
+    HighlightedTextSegment::HighlightedTextSegment(winrt::hstring const& textSegment, bool isHighlighted) :
+        _TextSegment(textSegment),
+        _IsHighlighted(isHighlighted)
+    {
+    }
+
+    HighlightedText::HighlightedText(Windows::Foundation::Collections::IObservableVector<winrt::TerminalApp::HighlightedTextSegment> const& segments) :
+        _Segments(segments)
+    {
+    }
+
+    DependencyProperty HighlightedTextControl::_textProperty = DependencyProperty::Register(
```

Comment:
I don't think I've authored a `DependencyProperty` yet, nor do I think we have any other ones we've authored in the Terminal codebase quite yet. Could you maybe litter some comments detailing how this works? It _seems_ like you set up this static `DependencyProperty` that's used to trigger `_onTextChanged` for the `HighlightedTextControl` who's `Text` changed, but I'm not entirely certain why we need to do this rather than just do this in the implementation of `HighlightedTextControl::Text(HighlightedText text)`